spark/src/main/scala/geotrellis/spark/reproject/TileRDDReproject.scala

```diff
@@ -119,34 +120,47 @@ object TileRDDReproject {
           }
         }
 
+    val reprojectSummary = matchReprojectRasterExtent(
+      metadata.crs, destCrs,
+      metadata.layout,
+      Some(metadata.bounds.asInstanceOf[KeyBounds[SpatialKey]])
```

Comment:
Here is a cast issue, can be resolved some how ~
```scala
metadata.bounds match {
        case KeyBounds(s, e) => Some(KeyBounds(s.getComponent[SpatialKey], e.getComponent[SpatialKey]))
        case _ => None
      }
```

Reply:
Thank you, good catch.